pyvo/registry/rtcons.py

```diff
@@ -0,0 +1,825 @@
+            parname = "fulltext{}".format(index)
+            parpatname = "fulltextpar{}".format(index)
+            self._fillers[parname] = word
+            self._fillers[parpatname] = '%'+word+'%'
+            for q in base_queries:
+                subqueries.append(q.format(**locals()))
+
+        self._condition = "ivoid IN ({})".format(
+            " UNION ".join(subqueries))
+
+
+class Author(Constraint):
+    """
+    A constraint for creators (“authors”) of a resource; you can use SQL
+    patterns here.
+
+    The match is case-sensitive.
+    """
+    _keyword = "author"
+
+    def __init__(self, name:str):
+        """
+
+        Parameters
+        ----------
+        name: str
+            Note that regrettably there are no guarantees as to how authors
+            are written in the VO.  This means that you will generally have
+            to write things like ``%Hubble%`` (% being “zero or more
+            characters” in SQL) here.
+        """
+        self._condition = "role_name LIKE {auth} AND base_role='creator'"
+        self._fillers = {"auth": name}
+
+
+class Servicetype(Constraint):
+    """
+    A constraint for for the availability of a certain kind of service
+    on the result.
+
+    The constraint normally is a custom keyword, one of:
+
+    * ``image``
+    * ``spectrum``
+    * ``scs`` (for cone search services)
+    * ``line`` (for SLAP services)
+    * ``table`` (for TAP services)
+
+    You can also pass in the standards' ivoid (which
+    generally looks like
+    ``ivo://ivoa.net/std/<standardname>`` and have to be URIs with
+    a scheme part in any case); note, however, that for standards
+    pyVO does not know about it will not build service instances for
+    you.
+
+    Multiple service types can be passed in; a match in that case
+    is for records having any of the service types passed in.
+
+    The match is literal (i.e., no patterns are allowed); this means
+    that you will not receive records that only have auxiliary
+    services, which is what you want when enumerating all services
+    of a certain type in the VO.  In data discovery (where, however,
+    you generally should not have Servicetype constraints), you
+    can use ``Servicetype(...).include_auxiliary_services()`` or
+    use registry.search's ``includeaux`` parameter; but, really, there
+    is little point using this constraint in data discovery in the first
+    place.
+    """
+    _keyword = "servicetype"
+
+    def __init__(self, *stds):
+        """
+
+        Parameters
+        ----------
+        *stds: tuple of str
+            one or more standards identifiers.  The constraint will
+            match records that have any of them.
+        """
+        self.stdids = set()
+
+        for std in stds:
+            if std in SERVICE_TYPE_MAP:
+                self.stdids.add(SERVICE_TYPE_MAP[std])
+            elif "://" in std:
+                self.stdids.add(std)
+            else:
```

Comment:
`std` is checked only in the keys of `SERVICE_TYPE_MAP`, not the values.
Previously, it checks both the keys and values.

So now, a search like:
```python
services = vo.regsearch(servicetype='conesearch')
```
doesn't work, where it previously used to work. `servicetype='scs'` works. I am not sure if this is the intended behavior. I think it will be useful to allow the search with both keys and values.